core/src/main/java/org/apache/druid/math/expr/Expr.java

```diff
@@ -356,42 +354,42 @@ public BindingDetails withScalarArguments(Set<Expr> scalarArguments)
      * Add set of arguments as {@link BindingDetails#arrayVariables} that are *directly* {@link IdentifierExpr},
      * else they are ignored.
      */
-    public BindingDetails withArrayArguments(Set<Expr> arrayArguments)
+    BindingDetails withArrayArguments(Set<Expr> arrayArguments)
     {
       Set<IdentifierExpr> arrayIdentifiers = new HashSet<>();
       for (Expr expr : arrayArguments) {
-        final IdentifierExpr isIdentifier = expr.getIdentifierExprIfIdentifierExpr();
-        if (isIdentifier != null) {
+        final boolean isIdentifierExpr = expr.getIdentifierExprIfIdentifierExpr() != null;
+        if (isIdentifierExpr) {
           arrayIdentifiers.add((IdentifierExpr) expr);
         }
       }
       return new BindingDetails(
           ImmutableSet.copyOf(Sets.union(freeVariables, arrayIdentifiers)),
           scalarVariables,
           ImmutableSet.copyOf(Sets.union(arrayVariables, arrayIdentifiers)),
-          hasInputArrays || arrayArguments.size() > 0,
+          hasInputArrays || !arrayArguments.isEmpty(),
```

Comment:
nit: I find older version better.

Reply:
Rationale for the change: https://rules.sonarsource.com/java/tag/clumsy/RSPEC-1155

Reply:
ok, rationale about isEmpty() in general being more efficient makes sense .